app/forms/search_form.rb

```diff
@@ -1,23 +1,49 @@
 class SearchForm
   include ActiveModel::Model
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveRecord::AttributeAssignment
+
+  Option = Struct.new(:value, :label)
+
+  attribute :search_string, :string
+  attribute :submitted_from, :date
+  attribute :submitted_to, :date
+  attribute :updated_from, :date
+  attribute :updated_to, :date
+  attribute :state, :string
+  attribute :office_code, :string
+  attribute :current_provider
 
-  def initialize(params)
-    @params = params.fetch(:search_form, {})
+  validate :at_least_one_attribute_set
+
+  def submitted?
+    !search_string.nil?
   end
 
-  def search_string
-    @params[:search_string]
+  def at_least_one_attribute_set
+    errors.add(:base, :no_attributes_set) if attributes.except('current_provider').values.none?(&:present?)
   end
 
-  def submitted?
-    @params.key?(:search_string)
+  def office_codes
+    [show_all] + current_provider.office_codes.map { Option.new(_1, _1) }
   end
 
-  def valid?
-    @params[:search_string].present?
+  def states
+    [show_all] + %i[
+      draft
```

Comment:
This is just all statuses no? Shouldn't this be a global array?

Reply:
Ah I didn't consider order, okay then. Just seems like we're bound to forget one some time.